### ISSUE TITLE
DataTable: RowGroup SummaryRow lost in a paginated lazy table if rows fit exactly with page size

### ISSUE THREAD

**Issue by sebsoftware** [primefaces/primefaces#9077]
### Describe the bug

If a datatable is paginated i.E. 10 rows, and the group rows are exactly fitting the first page the group header and the rows are rendered, but the summary row is not. The next page starts with the first row of the next group. 

### Expected behavior

Render the SummaryRow on the first page, even it exceeds the rows of the datatable

### PrimeFaces edition

Community

### PrimeFaces version

8 to 12

### Theme

Default, Nova

### JSF implementation

Mojarra

### JSF version

2.3

### Browser(s)

All

**Comment by tandraschko** [comment]
Please provide a reproducer

**Comment by sebsoftware** [comment]
Testing with the test-masters show that this happens only with lazy model.

1. Use the attached test.
2. Selected default page size is 10, no group summaries are show,
3. Change the page size and the summary rows are visible.
[primefaces-lazy-datatable-summaryrow.zip](https://github.com/primefaces/primefaces/files/9275346/primefaces-lazy-datatable-summaryrow.zip)

**Comment by tandraschko** [comment]
when our Renderer hits row=10, it checks if the a summary row must be encoded (DataTableRender#isInSameGroup)
so it compares row 10 and row 11 
but when setting the rowIndex to 11, somehow row 1 is restored, not 11

anyone has time for a creating a integration test?

**Comment by tandraschko** [comment]
after debugging further, when calling UIData#setRowIndex(10) (row = 11, it begins with 0):
it calls getDataModel().setRowIndex(10);
but
getDataModel().getRowIndex() returns 0


https://github.com/primefaces/primefaces/blob/master/primefaces/src/main/java/org/primefaces/model/LazyDataModel.java#L185
this makes 10 -> 0

**Comment by tandraschko** [comment]
NEVERMIND: i thought it works in 8.0 but not in 12.0

**Comment by tandraschko** [comment]
in general what should happen in the case when choose rows=15?

Page1:
- Header Amy Elsner
- 10 Customers
- Summary Row with 10 customers
- Header Anna Fali
- 5 Customers
Page2:
- Header Anna Fali
- 5 Customers
- Summary Row with 10 customers


or


Page1:
- Header Amy Elsner
- 10 Customers
- Summary Row with 10 customers
- Header Anna Fali
- 5 Customers
- Summary Row with 5 customers
Page2:
- Header Anna Fali
- 5 Customers
- Summary Row with 5 customers

**Comment by tandraschko** [comment]
i commited a prototype
maybe we should revisit why LazyDataModel needs to care about the pageSize and remove this in future

it would just be great if anyone could provide a integration test, based on the sample

**Comment by sebsoftware** [comment]
Find a quickly hacked test case for 8  attached. Code is not very well, but I hope it is good enough for testin
[primefaces-lazy-datatable-summaryrow-8.0.zip](https://github.com/primefaces/primefaces/files/9302430/primefaces-lazy-datatable-summaryrow-8.0.zip)
g

**Comment by sebsoftware** [comment]
Regarding your question about size 15, if the summary should be shown on a break or not.

Maybe it would be nice to have an option to show a summaryrow even if the group
is split across pages or only at the end of the group as a separator. 

But this is only nice to have, it is more important to don't loose a summary row :-)

**Comment by sebsoftware** [comment]
Just tested it with my application and it seems to work.
Good work!

**Comment by tandraschko** [comment]
@sebsoftware would you like to provide a intergration test based on your example? would be great contribution and it wont break again in future

**Comment by christophs78** [comment]
> @sebsoftware would you like to provide a intergration test based on your example? would be great contribution and it wont break again in future

You may build uppon https://github.com/primefaces/primefaces/blob/master/primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/datatable/DataTable016Test.java + https://github.com/primefaces/primefaces/blob/master/primefaces-integration-tests/src/main/webapp/datatable/dataTable016.xhtml. (DataTable016Test uses non-lazy datasource without paging. So you may add something like dataTable016_lazy.xhtml and so on.)

**Comment by sebsoftware** [comment]
I've tried to create an integration test. But integration-tests doesn't seem to work currently.

Even the example from the Readme is not working in the current master `mvn verify -Pintegration-tests,parallel-execution,mojarra-2.3 -Dit.test=DatePicker001Test` in my configuration.

I've attached current files, but I'm not able to test it. Maybe sthg is broken, or sthg in my environment isn't working.
At least building PrimeFaces works well. macOS Catalina. Maven 3.8.6.

[datatable-integration.zip](https://github.com/primefaces/primefaces/files/9397819/datatable-integration.zip)
.
